doc/rst/tools/mason/mason.rst

```diff
@@ -411,113 +608,122 @@ Mason uses the ``Spack`` package specification.
-    chplVersion = "1.18.0"
-    compopts = "--ccflags -Wno-enum-conversion --ccflags -Wno-strict-prototypes"
+`Mason-Registry <https://github.com/chapel-lang/mason-registry>`_.
 
-    [dependencies]
+A registry will be downloaded to ``$MASON_HOME/<name>`` by ``mason update``
+for each registry named in ``$MASON_REGISTRY`` if a registry at that location
+does not already exist.
 
-    [system]
-    lapack = "3.8.0"
-    openblas = "*"
+The registry consists of a hierarchy like the following:
 
+.. code-block:: text
 
-This ``Mason.toml`` is updated to reflect the necessary dependency information for Mason
-to build the package in the ``Mason.lock``. Mason gets external dependency information,
-in this case, from the ``.pc`` files found via the ``PKG_CONFIG_PATH``. The ``*`` means
-that Mason will grab the latest package available known to ``pkg-config`` in your system.
+ mason-registry/
+    Bricks/
+       Curl/
+          1.0.0.toml
+          2.0.0.toml
+       RecordParser/
+          1.0.0.toml
+          1.1.0.toml
+          1.2.0.toml
+       VisualDebug/
+          2.2.0.toml
+          2.2.1.toml
 
+Each versioned manifest file is identical to the manifest file in the top-level directory
+of the package repository, with one exception, a file path or URL pointing to the repository and revision
+in which the version is located.
 
-``Mason.lock``
+Continuing the example from before, the 'registry' ``0.1.0.toml`` would include the additional source field:
 
 .. code-block:: text
 
-   [root]
-   name = "cholesky"
-   compopts = "--ccflags -Wno-enum-conversion --ccflags -Wno-strict-prototypes"
-   version = "0.1.0"
-   chplVersion = "1.18.0..1.18.0"
+     [brick]
+     name = "MyPackage"
+     version = "0.1.0"
+     chplVersion = "1.16.0"
+     authors = ["Sam Partee <Sam@Partee.com>"]
+     source = "https://github.com/Spartee/MyPackage"
 
-   [system]
-        [system.lapack]
-        name = "lapack"
-        version = "3.8.0"
-        libs = "-L/usr/local/Cellar/lapack/3.8.0_1/lib -lapack"
-        include = "/usr/local/Cellar/lapack/3.8.0_1/include"
+     [dependencies]
+     curl = '1.0.0'
 
-        [system.openblas]
-        name = "openblas"
-        version = "0.3.1"
-        libs = "-L/usr/local/Cellar/openblas/0.3.1/lib -openblas"
-        include = "/usr/local/Cellar/openblas/0.3.1/include"
+Search the registry with ``mason search <query>``, which will list all packages
+(and their latest version) that contain ``<query>`` in their names (case-insensitive).
+If no query is provided, all packages in the registry will be listed.
 
+.. note::
 
-Use the ``mason system`` command for help with developing Mason packages with system
-package dependencies. Use ``mason system search <package>`` to lookup packages that are available
-on your system via ``pkg-config``. To view a package's ``.pc`` file, use the ``mason system pc <package>``
-command.
+    Packages will be listed regardless of their chplVersion compatibility.
```

Comment:
I did not realize this. We ought to provide functionality to restrict this search in the future.